route/build/src/kwt_route.f90

```diff
@@ -408,216 +409,218 @@ SUBROUTINE QROUTE_RCH(IENS,JRCH,    & ! input: array indices
- END SUBROUTINE QROUTE_RCH
-
-
-
-
-
-
-
-
+   USE reachparam
+   USE reachstate
+   USE reach_flux
+   implicit none
+   ! Input
+   INTEGER(I4B), INTENT(IN)                    :: IENS          ! ensemble member
+   integer(i4b), intent(in)                    :: ixOutlet      ! index of the outlet reach
+   INTEGER(I4B), INTENT(IN)                    :: JRCH          ! reach to process
+   REAL(DP), INTENT(IN)                        :: T0,T1         ! start and end of the time step (seconds)
+   INTEGER(I4B), INTENT(IN)                    :: MAXQPAR       ! maximum number of particles
+   INTEGER(I4B), INTENT(IN)                    :: LAKEFLAG      ! >0 if processing lakes
+   INTEGER(I4B), INTENT(IN), OPTIONAL          :: RSTEP         ! retrospective time step offset
+   ! output variables
+   integer(i4b), intent(out)                   :: ierr          ! error code
+   character(*), intent(out)                   :: message       ! error message
+   ! (1) extract flow from upstream reaches and append to the non-routed flow in JRCH
+   INTEGER(I4B)                                :: NUPS          ! number of upstream reaches
+   REAL(DP),DIMENSION(:),allocatable,SAVE      :: Q_JRCH        ! flow in downstream reach JRCH
+   REAL(DP),DIMENSION(:),allocatable,SAVE      :: TENTRY        ! entry time to JRCH (exit time u/s)
+   INTEGER(I4B)                                :: NQ1           ! # flow particles
+   ! (2) route flow within the current [JRCH] river segment     
+   INTEGER(I4B)                                :: ROFFSET       ! retrospective offset due to rstep
+   REAL(DP)                                    :: T_START       ! start of time step
+   REAL(DP)                                    :: T_END         ! end of time step
+   REAL(DP),DIMENSION(:),allocatable,SAVE      :: T_EXIT        ! time particle expected exit JRCH
+   LOGICAL(LGT),DIMENSION(:),allocatable,SAVE  :: FROUTE        ! routing flag .T. if particle exits
+   INTEGER(I4B)                                :: NQ2           ! # flow particles (<=NQ1 b/c merge)
+   ! (3) calculate time-step averages
+   INTEGER(I4B)                                :: NR            ! # routed particles
+   INTEGER(I4B)                                :: NN            ! # non-routed particles
+   REAL(DP),DIMENSION(2)                       :: TNEW          ! start/end of time step
+   REAL(DP),DIMENSION(1)                       :: QNEW          ! interpolated flow
+   ! (4) housekeeping
+   REAL(DP)                                    :: Q_END         ! flow at the end of the timestep
+   REAL(DP)                                    :: TIMEI         ! entry time at the end of the timestep
+   TYPE(FPOINT),allocatable,DIMENSION(:),SAVE  :: NEW_WAVE      ! temporary wave
+   LOGICAL(LGT),SAVE                           :: INIT=.TRUE.   ! used to initialize pointers
+   ! random stuff
+   CHARACTER(LEN=256)                          :: CMESSAGE      ! error message for downwind routine
 
```

Comment:
@nmizukami - can you add some inline comments on github describing where changes were made in the `QROUTE_RCH` subroutine. The indentation changes are making this tough to review.

Reply:
Sorry, I changed indentation so whole thing was changed....  I also changed case (from upper to lower) for some.   only a few real changes. I put inline comments.   1. change pointers to allocatable array, and 2. move MAXQPAR to input argument.